src/domain/auth/KukkuuHDSLoginCallbackhandler.tsx

```diff
@@ -0,0 +1,49 @@
+import { OidcClientError, User, LoginCallbackHandler } from 'hds-react';
+import * as Sentry from '@sentry/browser';
+import { useTranslation } from 'react-i18next';
+import { toast } from 'react-toastify';
+import { useNavigate } from 'react-router-dom';
+
+import useGetPathname from '../../common/route/utils/useGetPathname';
+
+function KukkuuHDSLoginCallbackHandler() {
+  const { t } = useTranslation();
+  const getPathname = useGetPathname();
+  const navigate = useNavigate();
+  const onSuccess = (user: User) => {
+    let redirectUrl = getPathname('/profile');
+    if (user.url_state) {
+      redirectUrl =
+        new URLSearchParams(user.url_state).get('next') ?? redirectUrl;
+    }
+    navigate(redirectUrl, { replace: true });
+  };
+
+  const onError = (error?: OidcClientError) => {
+    // eslint-disable-next-line
+    console.error(error);
+    if (!error) return;
+    if (
+      error.type === 'SIGNIN_ERROR' &&
+      error.message ===
+        'Current state (HANDLING_LOGIN_CALLBACK) cannot be handled by a callback'
+    ) {
+      // TODO: resolve this issue -- why is it raised?
+      // We don't want to handle this error,
+      // because it is raised during every login process,
+      // and it does not affect any how to the signin process.
```

Comment:
https://github.com/City-of-Helsinki/helsinki-design-system/blob/v3.7.0/packages/react/src/components/login/components/LoginCallbackHandler.tsx#L39

&

https://github.com/City-of-Helsinki/helsinki-design-system/blob/v3.7.0/packages/react/src/components/login/client/oidcClient.ts#L255-L276

are at least related...

Reply:
I actually do know what the reason for this is. It's a bug in the HDS-react package. It is not tested with so big application as the Kukkuu UI is: The Kukkuu UI has so many providers / container that makes the login provider to rerender, that it then raises those errenous signals that it should not raise. They promised to fix it.
```
      // The HANDLING_LOGIN_CALLBACK error is raised only
      // when we are using the <React.Strict>.
      // We don't want to handle this error,
      // because it is raised during every login process,
      // and it does not affect any how to the signin process.
      // When the <BrowserApp> is removed from the React.Strict -container,
      // The error won't be raised anymore. The HDS team has been noted about this issue.
```